server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -0,0 +1,924 @@
+        log.info("updateCompatibility called for group {}. New compatibility {}", group, compatibility);
+        return RETRY.runAsync(() -> store.getGroupEtag(namespace, group)
+                                         .thenCompose(pos -> {
+                                             return store.getGroupProperties(namespace, group)
+                                                         .thenCompose(prop -> {
+                                                             if (previousCompatibility == null) {
+                                                                 return store.updateCompatibility(namespace, group, pos, compatibility);
+                                                             } else {
+                                                                 if (previousCompatibility.equals(prop.getCompatibility())) {
+                                                                     return store.updateCompatibility(namespace, group, pos, compatibility);
+                                                                 } else {
+                                                                     throw new PreconditionFailedException("Conditional update failed");
+                                                                 }
+                                                             }
+                                                         });
+                                         })
+                                         .whenComplete((r, e) -> {
+                                             if (e == null) {
+                                                 log.info("Group {} updateCompatibility successful.", group);
+                                             } else {
+                                                 log.warn("getGroupProperties for group {} request failed with error", group, e);
+                                             }
+                                         }), executor);
+    }
+
+    /**
+     * Gets list of latest schema versions for all schemas registered in the group. Schemas representing different
+     * object types are identified by {@link SchemaInfo#type}.
+     *
+     * @param namespace  namespace for which the request is scoped to.
+     * @param group      Name of group.
+     * @param schemaType type of object as identified by {@link SchemaInfo#type}.
+     * @return CompletableFuture which holds list of latest schema versions for different schemas upon completion.
+     */
+    public CompletableFuture<List<SchemaWithVersion>> getSchemas(String namespace, String group, @Nullable String schemaType) {
+        Preconditions.checkArgument(group != null);
+        log.info("getSchemas called for group {}. New compatibility {}", group);
+
+        if (schemaType == null) {
+            return store.getLatestSchemas(namespace, group)
+                        .whenComplete((r, e) -> {
+                            if (e == null) {
```

Comment:
This kind of repeated code should be extracted out to a common method for reuse.

Reply:
answered above